### DISCUSSION TITLE
Hydra Integration with Kratos Cloud

### DISCUSSION THREAD

**Discussion by bilalsha** [ory/kratos#2293] [Q&A] [answered]
I am trying to make hydra work with Kratos cloud. I am using kratos-selfservice-ui-node for testing right now. When application sends request to `whoami` api I got unreadable message and after that I got `undefined` error as `Identity` object is not found in response.  

```
whoami >>> M�~����r� Bu�}׼	����M�
]�%�%*ǎh����                  �RZ@m��tS.�pR�&��D���_��y}3.���Ws���
      `(�1�/�!U�λiN��$��pE��*�W�7������<�1m���آ�:;���a�MH�#V��ĕX�q�n�u*�6f�n0���+xPh�j\I)��ox��`�� i�x���R쭓���MG�qM�u����*�aLkl]!�!�x���C�a^����]��]My��s�c�4 ��Ė3�mfy䣁-�7���z�w�ar��?Ĉ?NPA.Xx���0�f0��|���KN��BjJ(3N%����x�8�|��0<k�V4l�� �E�Uj$�K`�e#�hBO0jN�Ҋ4uF�8�"B����Rv����W���[�z�?�4�w���}Otp��8q""�e��$�B�%v4V���~�C�Wk�4�ЏǓ
TypeError: Cannot read property 'id' of undefined
    at /home/kratos-selfservice-ui-node/src/routes/hydra.ts:189:43
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

I also tried to send token in cookie, Authorization header but both gives same output. Below are the headers

```
headers {
  cookie: '_csrf=8IEALGKDylAoLB1UFUbVOnFB; connect.sid=s%3AX0eJEFtcCKZGMlrYsAXIOgcB_p1qKiDX.6DIw%2F1jOazX%2B3s9adsFgrz2iEesfh865fxDtRhQK808; csrf_token_eb7fd0866fd8188dbb4a7858c6e4eff0985bda5ae647e92e9e4ee08696e5ee5e=zgsiv5cgbsGGhOQc1QE72okbadR8BEhu6GFvs73H33A=; ory_session_hopefulcannonmxqwmpwo2g=MTY0NjY3ODM5NXxEdi1CQkFFQ180SUFBUkFCRUFBQVJfLUNBQUVHYzNSeWFXNW5EQThBRFhObGMzTnBiMjVmZEc5clpXNEdjM1J5YVc1bkRDSUFJRlZuYlZSek5FNWthR0UxZUdKMWVYbE1OSFJQWTJSek4xbENlVFV5T1UxWXwieQy6rWvvzDVcdMy2CPpGEXHA7nWYbUgOLwCC74-F6w==',
  Authorization: 'bearer MTY0NjY3ODM5NXxEdi1CQkFFQ180SUFBUkFCRUFBQVJfLUNBQUVHYzNSeWFXNW5EQThBRFhObGMzTnBiMjVmZEc5clpXNEdjM1J5YVc1bkRDSUFJRlZuYlZSek5FNWthR0UxZUdKMWVYbE1OSFJQWTJSek4xbENlVFV5T1UxWXwieQy6rWvvzDVcdMy2CPpGEXHA7nWYbUgOLwCC74-F6w=='
}
```

This is the code line in project https://github.com/ory/kratos-selfservice-ui-node/blob/hydra-integration-2021/src/routes/hydra.ts#L171

**Comment by gen1us2k** [comment]
Have you enabled your domain via [CNAME](https://www.ory.sh/docs/guides/custom-domains) feature of Ory Cloud? Could you please give an example of your domains setup?

You need to send a request with `ory_session_project_id` cookie `whoami` endpoint to return a session

**Comment by vinckr** [comment, marked as answer]
Hello @bilalsha 
The Ory Kratos and Ory Hydra integration just landed with Hydra 2.0,
its enabled out of the box on the Ory Network (formerly known as Ory Cloud),
see the [release notes](https://github.com/ory/hydra/releases/tag/v2.0.0) for more details!
